### PR TITLE
Issue 17

### PR DESCRIPTION
**Pull request by rick1e**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        python -m pip install --upgrade pip
+        pip install uv
+
+    - name: Set up environment with uv
+      run: |
+        uv venv
+        uv pip install --no-cache-dir -r requirements-dev.lock
+
+    - name: Add .venv to PATH
+      run: echo "$GITHUB_WORKSPACE/.venv/bin" >> $GITHUB_PATH
```
Comment (AshciR) [inline .github/workflows/ci.yml:33]: Why did we have to do this?

File: pyproject.toml [hunk 2]
```diff
@@ -9,7 +9,8 @@ authors = [
 dependencies = [
     "python-telegram-bot==20.8",
     "python-dotenv==1.0.1",
-    "SQLAlchemy==2.0.31"
+    "SQLAlchemy==2.0.31",
+    "pytest-xdist>=3.6.1",
```
Comment (AshciR) [inline pyproject.toml:13]: Typically you place pytest this in `dev_dependencies`

File: tests/conftest.py [hunk 3]
```diff
@@ -15,8 +15,11 @@ def db():
...
 
     session = SessionLocal()
-    yield session
-
-    # Close the session and drop all tables
-    session.close()
+    
+    try:
+        yield session
+    finally:
```
Comment (AshciR) [inline tests/conftest.py:21]: Good change

**Review by AshciR — CHANGES_REQUESTED** [review]
Change the dev dependencies

File: .github/workflows/ci.yml (hunk 1 above)
Comment (rick1e) [reply .github/workflows/ci.yml:33]: Idk that's how I saw it in the other one so I assumed it was just the standard set up for python

But let me know.

Comment (AshciR) [reply .github/workflows/ci.yml:33]: Ok, GITHUB actions require access to the `.venv`. 

Past me was smarter than present me

File: pyproject.toml (hunk 2 above)
Comment (rick1e) [reply pyproject.toml]: made the change

**Review by AshciR — APPROVED** [review]

**Pull request merged**